velox/exec/Window.cpp

```diff
@@ -187,6 +194,31 @@ void Window::createWindowFunctions() {
   }
 }
 
+// The supportRowLevelStreaming is designed to support 'rank' and
+// 'row_number' functions and the agg window function with default frame.
+bool Window::supportRowLevelStreaming() {
+  for (const auto& windowNodeFunction : windowNode_->windowFunctions()) {
+    const auto& functionName = windowNodeFunction.functionCall->name();
+    auto windowFunctionEntry =
+        exec::getWindowFunctionEntry(functionName).value();
+    if (windowFunctionEntry->processingUnit == ProcessingUnit::kPartition) {
+      return false;
+    }
+
+    const auto& frame = windowNodeFunction.frame;
+    bool isDefaultFrame =
+        (frame.startType == core::WindowNode::BoundType::kUnboundedPreceding &&
+         frame.endType == core::WindowNode::BoundType::kCurrentRow);
+    // Only support the agg window function with default frame.
+    if (!(functionName == "rank" || functionName == "row_number") &&
```

Comment:
We need to make this code independent of function names. Maybe we should add more options to ProcessingUnit for this.

Reply:
@aditi-pandit Add `ignoreFrame ` option in ProcessingUnitMetadata.